web/src/frontend/classic/stores/Documentation.ts

```diff
@@ -0,0 +1,80 @@
+import { writable } from 'svelte/store';
+
+export type DocCategory = {
+    name: string;
+    description: HTMLElement;
+    sections: DocSection[];
+}
+export type DocSection = {
+    name: string,
+    content: HTMLElement
+}
+
+export const documentation = writable<DocCategory[]>([]);
+
+/**
+ * Creates a slide section by extracting HTML elements from the given content element until a specified split element is encountered.
+ * @param content - The content element from which to extract the slide section.
+ * @param split - The element name at which to stop extracting and split the slide section.
+ * @returns The created slide section element.
+ */
+function slideElement(content:HTMLElement, split:string[]) : HTMLElement {
+    const slide = document.createElement("div");
+    let nextSibling = content.nextSibling;
+    while (nextSibling && !split.includes(nextSibling.nodeName)) {
+        slide.appendChild(nextSibling.cloneNode(true));
+        nextSibling = nextSibling.nextSibling;
+    }
+    return slide;
+}
+
+(async function () {
+    const url = 'https://hakuneko.download/docs/haruneko/';
+    // TODO Documentation should be cached using a service worker
+    const cacheKey = `cache:${url}`;
+    const cacheTimestamp = localStorage.getItem(`${cacheKey}:timestamp`);
+    if (cacheTimestamp) {
```

Comment:
This cache strategy has some drawbacks, the preferred strategy would be online first and fallback to cache if loading online docs failed (this would also be the strategy of choice for a service worker).

Reply:
This was intentional so that there is no request to the server each time the app is started (as the guides are displayed on the front page). Just once a week.